server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MessageAppender.java

```diff
@@ -56,27 +58,32 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(MessageAppender.class);
 
     private final MailboxManager mailboxManager;
+    private final MessageIdManager messageIdManager;
     private final AttachmentManager attachmentManager;
     private final MIMEMessageConverter mimeMessageConverter;
 
     @Inject
-    public MessageAppender(MailboxManager mailboxManager, AttachmentManager attachmentManager, MIMEMessageConverter mimeMessageConverter) {
+    public MessageAppender(MailboxManager mailboxManager, MessageIdManager messageIdManager, AttachmentManager attachmentManager, MIMEMessageConverter mimeMessageConverter) {
         this.mailboxManager = mailboxManager;
+        this.messageIdManager = messageIdManager;
         this.attachmentManager = attachmentManager;
         this.mimeMessageConverter = mimeMessageConverter;
     }
 
     public MessageFactory.MetaDataWithContent appendMessageInMailbox(CreationMessageEntry createdEntry,
-                                                                     MessageManager mailbox,
+                                                                     List<MailboxId> targetMailboxes,
                                                                      MailboxSession session) throws MailboxException {
         ImmutableList<MessageAttachment> messageAttachments = getMessageAttachments(session, createdEntry.getValue().getAttachments());
         byte[] messageContent = mimeMessageConverter.convert(createdEntry, messageAttachments);
         SharedByteArrayInputStream content = new SharedByteArrayInputStream(messageContent);
         Date internalDate = Date.from(createdEntry.getValue().getDate().toInstant());
 
         boolean notRecent = false;
-
+        MessageManager mailbox = mailboxManager.getMailbox(targetMailboxes.get(0), session);
         ComposedMessageId message = mailbox.appendMessage(content, internalDate, session, notRecent, getFlags(createdEntry.getValue()));
+        if (targetMailboxes.size() > 0) {
```

Comment:
and then you check that size is > 0? It smells bad.

Reply:
Oups, that was supposed to be a > 1 (to avoid needless setInMailboxes calls)